examples/spark2/src/main/scala/org/apache/carbondata/benchmark/ConcurrentQueryBenchmark.scala

```diff
@@ -0,0 +1,631 @@
+      timeArray(i) =
+        ((results.get(i).get().startTime - start) / (1000.0 * 1000),
+          (results.get(i).get().endTime - start) / (1000.0 * 1000),
+          (results.get(i).get().endTime - results.get(i).get().startTime) / (1000.0 * 1000))
+    }
+    val timeArraySorted = timeArray.sortBy(x => x._1)
+    val timeArrayString = timeArraySorted.map { e =>
+      e._1.formatted("%.3f") + ",\t" + e._2.formatted("%.3f") + ",\t" + e._3.formatted("%.3f")
+    }
+    val saveArray = Array(sql, "startTime, endTime, runtime, measure time by the microsecond",
+      s"${timeArrayString.length}")
+      .union(timeArrayString)
+    val rdd = spark.sparkContext.parallelize(saveArray, 1)
+    rdd.saveAsTextFile(filePath)
+  }
+
+  /**
+   * print out results
+   *
+   * @param results        Results
+   * @param sql            query sql
+   * @param tasksStartTime tasks start time
+   */
+  def printResults(results: util.List[Future[Results]], sql: String = "", tasksStartTime: Long) {
+    val timeArray = new Array[Double](results.size())
+    val sqlResult = results.get(0).get().sqlResult
+    for (i <- 0 until results.size()) {
+      results.get(i).get()
+    }
+    for (i <- 0 until results.size()) {
+      timeArray(i) = results.get(i).get().time
+    }
+    val sortTimeArray = timeArray.sorted
+
+    // the time of 90 percent sql are finished
+    val time90 = ((sortTimeArray.length) * 0.9).toInt - 1
+    // the time of 99 percent sql are finished
+    val time99 = ((sortTimeArray.length) * 0.99).toInt - 1
+    print(
+      "Min: " + sortTimeArray.head.formatted("%.3f") + " s," +
+        "\tMax: " + sortTimeArray.last.formatted("%.3f") + " s," +
+        "\t90%: " + sortTimeArray(time90).formatted("%.3f") + " s," +
+        "\t99%: " + sortTimeArray(time99).formatted("%.3f") + " s," +
+        "\tAvg: " + (timeArray.sum / timeArray.length).formatted("%.3f") + " s," +
+        "\t\tCount: " + results.get(0).get.count +
+        "\t\t\t\t" + sql +
+        "\t" + sqlResult.mkString(",") + "\t")
+  }
+
+  /**
+   * save result after finishing each task/thread
+   *
+   * @param time      each task time of executing query sql  and with millis time
+   * @param sqlResult query sql result
+   * @param count     result count
+   * @param startTime task start time with nano time
+   * @param endTime   task end time with nano time
+   */
+  case class Results(
+      time: Double,
+      sqlResult: Array[Row],
+      count: Int,
+      startTime: Long,
+      endTime: Long)
+
+
+  class QueryTask(spark: SparkSession, query: String)
+    extends Callable[Results] with Serializable {
+    override def call(): Results = {
+      var result: Array[Row] = null
+      val startTime = System.nanoTime()
+      val rt = time {
+        result = spark.sql(query).collect()
+      }
+      val endTime = System.nanoTime()
+      if (resultIsEmpty) {
+        Results(rt, Array.empty[Row], count = result.length, startTime, endTime)
```

Comment:
it is better not to return the result, it will keep the result in memory thus GC a lot

Reply:
resultIsEmpty default  is true, so it will not return result defaultly.  If  user want know the result, such validate the compute result, user can set the resultIsEmpty  as false. I have tested it, GC acceptable for small test size.